grails-app/conf/UrlMappings.groovy

```diff
@@ -255,6 +255,23 @@ class UrlMappings {
             action = [GET:"currencies"]
         }
 
+        // Invoice API
+        "/api/invoices/$id/invoiceItems"(parseRequest: true) {
+            controller = "invoiceApi"
+            action = [GET: "getInvoiceItems"]
+        }
+
+        "/api/invoices/$id/removeItem"(parseRequest: true) {
+            controller = "invoiceApi"
+            action = [DELETE: "removeItem"]
+        }
+
+        "/api/invoices/$id/updateItems"(parseRequest: true) {
```

Comment:
We should be using the request method to tell us what action to perform. The API should just be /api/invoices/:id/items

Reply:
Thanks, that's right. I have updated PR.